### PR TITLE
Migrate @storybook/addon-ondevice-backgrounds to Typescript

### PR DESCRIPTION
**Pull request by lonyele**
## What I did
I migrated addon-ondevice-backgrounds to Typescript

## How to test

- Is this testable with Jest or Chromatic screenshots? No
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

Tracking Issue: #5030

### CODE REVIEW

File: addons/ondevice-backgrounds/src/container.tsx [hunk 1]
```diff
@@ -32,18 +41,3 @@ export default class Container extends React.Component {
     );
   }
 }
-
```
Comment (gaetanmaisse) [inline addons/ondevice-backgrounds/src/container.tsx]: W

File: addons/ondevice-backgrounds/src/container.tsx [hunk 2]
```diff
@@ -32,18 +41,3 @@ export default class Container extends React.Component {
...
-Container.propTypes = {
```
Comment (gaetanmaisse) [inline addons/ondevice-backgrounds/src/container.tsx]: Why do you remove these `propTypes` and not the one of `Swatch` component?

**Review by gaetanmaisse — APPROVED** [review]

File: addons/ondevice-backgrounds/src/container.tsx (hunk 2 above)
Comment (lonyele) [reply addons/ondevice-backgrounds/src/container.tsx]: As I know of, React class component can't be used with PropTypes? I get this error `Property 'propTypes' does not exist on type 'typeof Container`  So it seems like only working with functional component(`Swatch` is a functional component) I wanted to leave `PropTypes` for runtime check but I couldn't do it. Is there any other way to do it?

Comment (gaetanmaisse) [reply addons/ondevice-backgrounds/src/container.tsx]: True! I already faced that problem before... and finally, I did not found a proper solution. Either cast `Container` as any to define propTypes or remove them, I picked the 2nd one just like you did.

Comment (lonyele) [reply addons/ondevice-backgrounds/src/container.tsx]: Yeah I feel like it's weird but it seems like it is working as intended

**Pull request merged**